packages/integration_test/lib/integration_test.dart

```diff
@@ -195,28 +195,29 @@ class IntegrationTestWidgetsFlutterBinding extends LiveTestWidgetsFlutterBinding
     results[description] ??= _success;
   }
 
-  vm.VmService _vmService;
+  vm.VmService? _cachedVmService;
+  Future<vm.VmService> get _vmService async {
+    if (_cachedVmService == null) {
+      final developer.ServiceProtocolInfo info =
+          await developer.Service.getInfo();
+      assert(info.serverUri != null);
+      _cachedVmService = await vm_io.vmServiceConnectUri(
+        'ws://localhost:${info.serverUri!.port}${info.serverUri!.path}ws',
+      );
+    }
+    return _cachedVmService!;
+  }
 
   /// Initialize the [vm.VmService] settings for the timeline.
   @visibleForTesting
   Future<void> enableTimeline({
     List<String> streams = const <String>['all'],
-    @visibleForTesting vm.VmService vmService,
+    @visibleForTesting vm.VmService? vmService,
   }) async {
-    assert(streams != null);
+    assert(streams != null); // ignore: unnecessary_null_comparison
```

Comment:
This line seems no longer necessary when we have null-safety.

Reply:
It's still useful when not in sound null safety e.g. importing a non nnbd package

Reply:
nit: does it make sense to merge to asserts?

Reply:
@nturgut I'm not sure I'm understanding what merge asserts would be in this context...

Reply:
it's very nit :) 

assert(streams !=null && streams. isNotEmpty, 'Non empty stream is required')